packages/components/src/Gallery/Gallery.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from "react";
+import styles from "./Gallery.css";
+import { GalleryProps } from "./GalleryTypes";
+import { LightBox } from "../LightBox";
+import { FormatFile } from "../FormatFile";
+import { Button } from "../Button";
+
+export function Gallery({ files, size = "default", max }: GalleryProps) {
+  const images = files
+    .filter(file => file.type.startsWith("image/"))
+    .map(file => {
+      return { title: file.name, url: file.src };
+    });
+
+  const [lightboxOpen, setLightboxOpen] = useState(false);
+  const [lightboxIndex, setLightboxIndex] = useState(0);
+  const [displayPastMax, setDisplayPastMax] = useState(false);
+
+  return (
+    <>
+      <div className={size === "large" ? styles.galleryLarge : styles.gallery}>
+        {files.map((file, index) => {
+          if (max && !displayPastMax && index + 1 > max) {
+            return <></>;
+          }
+          return (
+            <div key={file.key}>
+              <FormatFile
+                file={{ ...file, src: () => Promise.resolve(file.src) }}
+                display="compact"
+                displaySize={size}
+                onClick={() => {
+                  handleThumbnailClicked(index);
+                }}
+                onDelete={file.onDelete}
```

Comment:
```suggestion
                onDelete={handleDelete(file)}
```
Sorry about the miscommunication earlier. Would it be possible for the `Gallery` to accept a function called `onDelete`. Then, for each individual `file` that was passed in, it will trigger that function on the file? So the code would look like what I suggested, but the `Gallery` would then require another input `prop`, the `handleDelete` method